### PR TITLE
`CreatorPage`: initial implementation

### PR DESCRIPTION
**Pull request by roy-guggenheim**

### CODE REVIEW

File: app/src/utils/utils.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
+/**
```
Comment (mkimi-lightricks) [inline app/src/utils/utils.ts]: I'm against having files with such broad responsibilities as they become a garbage collection of the entire code base.

Instead of having one `utils` file, I suggest splitting the methods in this file into individual files by theme:
- The `randomInt` method can be placed into the `math.ts` file as it essentially extends `Math`.
- `capitalizeFirstLetter` can be moved into `strings.ts` (for the same reason as above).
- `getRandomElements` effectively extends the `Array` functionality, hence it should be placed into the `array.ts` file.

File: app/src/components/Tabs/Tabs.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import React, { useState } from "react";
+import { capitalizeFirstLetter } from "../../utils/utils";
+import "./Tabs.scss";
+
+export type Tab = 'overview' | 'posts';
+
+type TabsProps = {
+    tabs: Tab[];
+    currentTab: Tab;
+    onTabChange: (tab: Tab) => void;
```
Comment (mkimi-lightricks) [inline app/src/components/Tabs/Tabs.tsx]: This API ceases to function when someone attempts to instantiate the same tab twice, as in: `['posts', 'posts']`. For instance, let's say `currentTab` is `post`, but you can't tell which one it is.

enhance this API to accommodate this particular scenario.

File: app/src/components/Post/Post.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+export type PostProps = {
+    id: string;
+    body: string;
+    comments: Comment[];
+}
+
+function Post(props: PostProps) {
+    const [isExpanded, setIsExpanded] = useState(false);
```
Comment (mkimi-lightricks) [inline app/src/components/Post/Post.tsx]: This code functions, but it is fragile. It will break whenever UX introduces an additional method to preview the comments, such as previewing the most popular comments. You will need to broaden the scope of `isExpanded` to encompass the new *most popular comments* scenario. 

The correct approach to this type of modeling is to utilize Typescript's 'union types'. An example would be: `type Preview = 'collapsed' | 'expanded'`. This allows for flexibility in expansion. (Regrettably, developers often default to collapsing every two-state variable into a boolean, which isn't correct).

In conclusion, incorporate a union type for the comments preview and utilize it in place of the `isExpanded`.

File: app/src/api/usersApi.ts [hunk 4]
```diff
@@ -28,35 +31,95 @@ export type User = {
...
+
+    return transformUserData(userData, overviewText);
+}
+
+/**
+ * Fetches a list of posts
+ * @param numPosts Number of posts to fetch
+ * @returns List of posts
+ */
+export async function fetchPosts(numPosts: number): Promise<PostProps[]> {
```
Comment (mkimi-lightricks) [inline app/src/api/usersApi.ts:64]: Since this is just a toy example, it can remain as it is. However, normally, this API would be incorrect. You should always design APIs that support pagination, similar to the one we created here: `fetchUsers(startIndex: number, length: number)`.

**Review by mkimi-lightricks — CHANGES_REQUESTED** [review]
partial round, I haven't review the last commit.

File: app/src/components/CreatorPage/CreatorPage.tsx [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+                                        </div>
+                                        <p>{userData.email}</p>
+                                    </div>
+                                    <div className="edit-button-wrapper">
+                                        <LTButton className="edit-button"
+                                            label={"Edit"}
+                                            onClick={props.onEditClick} />
+                                    </div>
+                                </div>
+                            </div>
```
Comment (mkimi-lightricks) [inline app/src/components/CreatorPage/CreatorPage.tsx]: Extract this into a component on its own, as the structure of the creator page becomes too complicated.

File: app/src/components/CreatorPage/CreatorPage.tsx [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+        }
+        fetchUserData();
+    }, [props.userId])
+
+    if (!userData) {
+        return <div className="loading">Loading...</div>;
+    }
+
+    return (
+        <div className={`page ${currentTab}-tab`}>
```
Comment (mkimi-lightricks) [inline app/src/components/CreatorPage/CreatorPage.tsx]: The mapping from the `Tab` type to a corresponding class in `CreatorPage.scss` is currently implicit. This means that if we introduce a new variant to the `Tab` type, `${currentTab}-tab` may cease to function without generating any error message. 

To rectify this, we should make the mapping explicit. This can be achieved by using a switch-case structure, as illustrated below:

```ts
function componentClass(tab: Tab): string {
  switch (tab) {
    case "overview":
      return "overview-tab";
    case "posts":
      return "posts-tab";
  }
}
```
For best practices, whenever you're using union types (for instance, `export type FooType = "foo" | "bar" | "baz";`), it's recommended to also use switch-case structures in the code for variables of this type: `let foo: FooType`;

**Review by mkimi-lightricks — CHANGES_REQUESTED** [review]
round.

**Review by mkimi-lightricks — APPROVED** [review]
fix squash and merge.
